src/web/js/dashboard/StudentDashboard.js

```diff
@@ -1,57 +1,63 @@
 import React, { Component } from 'react';
 import GoogleAPI from './GoogleAPI.js';
-import {CLIENT_ID, FILE_EXT, APP_NAME, API_KEY} from './config.js';
 import File from './File';
+import ReactDOM from 'react-dom';
+import '../../css/dashboard/index.css';
+
+import 'react-mdl/extra/material.css';
+import 'react-mdl/extra/material.js';
+import { Layout, Header, Textfield, Navigation, Content, Button, HeaderRow, HeaderTabs, Tab, Spinner } from 'react-mdl';
 
 class StudentDashboard extends Component {
   constructor() {
     super();
 
-    this.state = {signedIn: false, files: [], activeTab: 'recent-files', newFileName: ''};
+    this.state = {
+      activeTab: 0,
+      signedIn: false,
+      recentFiles: [],
+      newFileName: '',
+      fileSpinnerActive: true,
+      templateFiles: [
+        {name: 'Sort a List.arr', id: '0B32bNEogmncOTEJjQ1VicHdlYmc'},
+        {name: 'Compute a Derivative.arr', id: '0B32bNEogmncOWU9OWW5MSFlHSDQ'},
+        {name: 'Land a plane.arr', id: '0B32bNEogmncONnZNU2JsUnRVRG8'},
+        {name: 'Play 2048.arr', id: '0B32bNEogmncOMTg5T2plV19LX0k'}
+      ]
+    };
 
     this.api = new GoogleAPI();
-    this.api.load().then((resp) => {
+    this.api.load().then(() => {
       this.setState({signedIn: true});
       this.updateRecentFiles();
     });
   }
 
-  handleSignInClick = (event) => {
-    this.api.signIn().then((resp) => {
+  componentDidUpdate = (prevProps, prevState) => {
+    if (this.state.recentFiles != prevState.recentFiles) {
```

Comment:
I'd prefer `!==` here since I think this really should do object identity comparison, and it avoids accidental stringified comparison errors with `undefined`.